include/openenclave/debugrt/host.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * @file host.h
+ *
+ * This file defines the structures and functions used by the
+ * host-side debugger runtime.
+ */
+#ifndef _OE_DEBUGRT_HOST_H
+#define _OE_DEBUGRT_HOST_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/internal/defs.h> // For OE_STATIC_ASSERT
+
+OE_EXTERNC_BEGIN
+
+#ifdef __linux__
+typedef int32_t wide_char_t;
+#else
+typedef int16_t wide_char_t;
+#endif
+
+#define OE_DEBUG_ENCLAVE_MAGIC 0xabc540ee14fa48ce
+
+typedef struct _debug_enclave_t
+{
+    uint64_t magic;
+
+    const char* path;
+    const wide_char_t* wpath;
+
+    const void* base_address;
+
+    struct _sgx_tcs** tcs;
+    uint64_t num_tcs;
+
+    bool debug;
+    bool simulate;
+
+    struct _debug_enclave_t* next;
```

Comment:
Nonblocking: Consider making the data structure forward compatible:
- It might be useful to explicitly version this data structure (i.e. add `uint32_t version` field) otherwise versioning might depend on handling multiple `magic` values. 
- It seems likely that `next` will be a stable part of the structure and should be shifted prior to the flags that may grow (e.g. new flags) or deprecated (e.g. `simulate` may go away)
- It may be desirable to take an extensible data structure (e.g. bit flags with unused bit fields) instead of individual bools

Reply:
All above suggestions incorporated.